x/stake/pool_test.go

```diff
@@ -257,21 +257,19 @@ func TestCandidateRemoveShares(t *testing.T) {
 // TODO Make all random tests less obfuscated!
 
 // generate a random candidate
-func randomCandidate(r *rand.Rand) Candidate {
+func randomCandidate(r *rand.Rand, i int) Candidate {
```

Comment:
i updated randomCandidate() and randomSetup() to better accommodate random testing of candidates by using an index for the candidates.